autocertkit/ack_cli.py

```diff
@@ -421,22 +428,29 @@ def main(config, test_run_file):
     # Start Logger
     utils.init_ack_logging(session)
 
+    utils.log.info("Options: %s" % config)
+
     # Pre checks before running tests
     pre_flight_checks(session, config)
 
     config['xs_version'] = utils.get_xenserver_version(session)
     config['xcp_version'] = utils.get_xcp_version(session)
 
     generate_test_config(session, config, test_run_file)
-    # Logout of XAPI session anyway - the test runner will create a new session
-    # if needed. (We might only be generating).
-    session.logout()
 
     if 'generate' in config:
         # Generate config file only
         utils.log.info("Test file generated")
+        session.logout()
         return "OK"
 
+    # cleanup in case previous run did not complete entirely
+    utils.pool_wide_cleanup(session)
```

Comment:
During SR-IOV test running, it may let host reboot (e.g. when VF enabled). After host rebooted, test runner executes again, then pool_wide_cleanup should not be run, otherwise you know SR-IOV test environment will be destroyed exceptionally and could not continue to run rest steps (e.g. VF assignment).
Here, the utils.pool_wide_cleanup(session) is to cleanup the previous run of whole ACK, not for certain test case. And the place of ack_cli.py is right where of ACK launch.

I also added cleanup after one test case done in testbase.py:
```
            # cleanup occurs only when current test really done
            if rec['status'] == 'done':
                try:
                    pool_wide_cleanup(self.session)
```